From 3ed3e2229e98ae873aef71796de6f802393d6543 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:44 -0400
Subject: [ethernet] i40e/i40evf: set proper default for ITR registers

Message-id: <1409658042-22689-104-git-send-email-sassmann@redhat.com>
Patchwork-id: 90298
O-Subject: [RHEL7.1 PATCH 103/281] i40e/i40evf: set proper default for ITR registers
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Ethtool consistently reports 0 values for our ITR settings because
we never actually set them. Fix this by setting the default values
to the specified default values.

Change-ID: I2832406a66f7140f2b1230945d6ff6cbf77467c8
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit ca99eb997a30c445a1f01dd75012bbac744004d9)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 580d559..c964af9 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -6368,6 +6368,10 @@ static int i40e_sw_init(struct i40e_pf *pf)
       I40E_FLAG_MSIX_ENABLED    |
       I40E_FLAG_RX_1BUF_ENABLED;
 
+ /* Set default ITR */
+ pf->rx_itr_default = I40E_ITR_DYNAMIC | I40E_ITR_RX_DEF;
+ pf->tx_itr_default = I40E_ITR_DYNAMIC | I40E_ITR_TX_DEF;
+
  /* Depending on PF configurations, it is possible that the RSS
   * maximum might end up larger than the available queues
   */
-- 
1.7.1